packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -1,9 +1,122 @@
 import * as codebuild from '@aws-cdk/aws-codebuild';
 import * as iam from '@aws-cdk/aws-iam';
+import * as kms from '@aws-cdk/aws-kms';
 import * as sfn from '@aws-cdk/aws-stepfunctions';
 import * as cdk from '@aws-cdk/core';
 import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
 
+/**
+ * Status for log configuration
+ */
+export enum BuildLogConfigStatus {
+  /**
+   * The status 'ENABLED'
+   */
+  ENABLED = 'ENABLED',
+
+  /**
+   * The status 'DISABLED'
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * The log config of CloudWatch logs
+ */
+export interface CloudWatchLogsConfig {
+  /**
+   * The current status of the logs in Amazon CloudWatch Logs for a build project.
+   */
+  readonly status: BuildLogConfigStatus,
+
+  /**
+   * The group name of the logs in Amazon CloudWatch Logs.
+   *
+   * @default - the name of log group is not specified
+   */
+  readonly groupName?: string,
+
+  /**
+   * The prefix of the stream name of the Amazon CloudWatch Logs.
+   *
+   * @default - the name of log stream is not specified
+   */
+  readonly streamName?: string,
+}
+
+/**
+ * Information about S3 logs for a build project.
+ */
+export interface S3LogsConfig {
+  /**
+   * The current status of the S3 build logs.
+   */
+  readonly status: BuildLogConfigStatus,
+
+  /**
+   * Set to true if you do not want your S3 build log output encrypted.
+   *
+   * @default - by default S3 build logs are encrypted.
+   */
+  readonly encryptionDisabled?: boolean,
+
+  /**
+   * The ARN of an S3 bucket and the path prefix for S3 logs.
+   *
+   * @default - the location is not specified
+   */
+  readonly location?: string,
+}
+
+/**
+ * The log config of build of CodeBuild project
+ */
+export interface BuildLogConfig {
+  /**
+   * CloudWatch logs configuration.
+   *
+   * @default - no CloudWatch logs is specified
+   */
+  readonly cloudWatchLogs?: CloudWatchLogsConfig,
+
+  /**
+   * Information about logs built to an S3 bucket for a build project.
+   *
+   * @default - no s3 log is specified
+   */
+  readonly s3Logs?: S3LogsConfig,
+}
+
+/**
+ * Environment type for Linux Docker images
+ */
+export enum ImageType {
```

Comment:
Please remove this enum. This should be modeled with `IBuildImage` from the CodeBuild module.